src/components/dropdown/dropdown.svelte

```diff
@@ -0,0 +1,254 @@
+    // tab/shift+tab.
+    let dir = 0
+    if (e.code == 'ArrowUp') dir -= 1
+    if (e.code === 'ArrowDown') dir += 1
+    if (dir === 0) return
+
+    // First, find the currently focusedIndex. If no option is selected, we'll
+    // select the first option. Otherwise, we select the next/previous item (and
+    // wrap around).
+    let focusedIndex = options.findIndex((e) => e.matches(':focus-within'))
+    if (focusedIndex === -1) {
+      focusedIndex = 0
+    } else {
+      focusedIndex += dir
+      if (focusedIndex < 0) focusedIndex = options.length - 1
+      if (focusedIndex >= options.length) focusedIndex = 0
+    }
+
+    ;(options[focusedIndex] as any)?.focus()
+    e.preventDefault() // preventDefault, so we don't accidentally scroll
+  }
+</script>
+
+<div class="leo-dropdown">
+  <Control bind:disabled bind:required bind:size>
+    <slot name="label" slot="label" />
+    <slot name="left-icon" slot="left-icon" />
+    <button
+      bind:this={button}
+      class="click-target"
+      {disabled}
+      on:click={(e) => (isOpen = !isOpen)}
+    >
+      {#if value}
+        <slot name="value" {value}>
+          <span class="value">{value}</span>
+        </slot>
+      {:else}
+        <slot name="placeholder">
+          <span class="placeholder">{placeholder}</span>
+        </slot>
+      {/if}
+    </button>
+    <slot name="right-icon" slot="right-icon">
+      <div class="indicator" class:open={isOpen}>
+        <Icon name="arrow-small-down" />
+      </div>
+    </slot>
+    <div class="menu" slot="after">
+      {#if isOpen}
+        <div
+          class="leo-dropdown-popup"
+          transition:scale={{ duration: 60, start: 0.8 }}
+          use:clickOutside={(e) => (isOpen = false)}
+          on:keypress={(e) => {
+            if (e.code !== 'Enter' && e.code !== 'Space') return
+            selectOption(e)
+          }}
+          on:click={selectOption}
+          bind:this={popup}
+        >
+          <slot />
+        </div>
+      {/if}
+    </div>
+  </Control>
+</div>
+
+<svelte:window on:keydown={changeSelection} />
+
+<style lang="scss">
+  .leo-dropdown {
+    --dropdown-gap: var(--leo-dropdown-gap, var(--leo-spacing-8));
+    --transition-duration: var(--leo-dropdown-transition-duration, 0.12s);
+
+    cursor: pointer;
+    -webkit-tap-highlight-color: transparent;
+
+    button {
+      all: unset;
+    }
+  }
+
+  .leo-dropdown .click-target {
+    flex: 1;
+  }
+
+  .leo-dropdown .value {
+    width: 100%;
+  }
+
+  .leo-dropdown .placeholder {
+    color: var(--leo-color-text-secondary);
+  }
+
+  .leo-dropdown .menu {
+    position: relative;
+    z-index: 1000;
+  }
+
+  .leo-dropdown .indicator {
+    transition: transform var(--transition-duration) ease-in-out;
+
+    &.open {
+      transform: rotate(-180deg);
+    }
+  }
+
+  .leo-dropdown .leo-dropdown-popup {
+    background: var(--leo-color-container-background);
+    box-shadow: var(--leo-effect-elevation-03);
+    position: absolute;
```

Comment:
This won't break out of `overflow: hidden` containers, right? I'm not saying you need to fix in this PR, but it has caused problems in the past so perhaps we can look at it in a follow-up, maybe using https://developer.mozilla.org/en-US/docs/Web/API/Popover_API.

Another issue that this can cause is knowing whether to open down or up and not having the popup being cut off by the viewport.

Reply:
Indeed it will not. That's a rabbit hole I wanted to avoid at first :laughing: 

The Popover API sounds great (and actually seems to work with the #experimental-web-platform-features flag) but I worry it doesn't have broad enough support yet.